### PR TITLE
MM-59529 Set Channel/Team Admin permissions if All members get set

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
When channel moderated permissions were implemented, the Team/Channel Admins were given the permission by default. This would allow them to still access channels that had been moderated. However, it is now possible to get into a state when the Team/Channel Admin does not have that permission, but the "All members" section does have that permission. When a channel is moderated, the Team/Channel Admin lose access to that channel. A Team/Channel Admin should not be affected by Channel Moderation. 

Since these permissions are considered inherited, if `All members` is given access to a Moderated Permission, it is automatically re-added to the Team/Channel Admin.

Also added a permission migration to add the `UploadFile` permission to wherever the `CreatePost` permission exists, since both are required to be able to full create a post.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-59529

#### Release Note

```release-note
Fixed an issue where Team and Channel Admins could lose the ability to Create a post in a moderated channel.
```

### CODE REVIEW

**Review by sbishel — COMMENTED** [review]
`permission_system_schema_settings` and `permission_team_schema_settings` should really be brought together. They are very similar, however, with enough differences to make it a challenge.

I attempted to share the `TogglePermission` method, but because of difference is objects uses, I found myself going in deeper and deeper. Finally, decided it was beyond the scope of this issue.

**Review by harshilsharma63 — APPROVED** [review]

File: webapp/channels/src/utils/constants.tsx [hunk 1]
```diff
@@ -1391,6 +1391,16 @@ export const DefaultRolePermissions = {
     ],
 };
 
+export const ModeratedPermissions = [
```
Comment (BenCookie95) [inline webapp/channels/src/utils/constants.tsx:1394]: 1/5 I think it would be useful to leave a comment describing why we need this constant

**Review by BenCookie95 — APPROVED** [review]
LGTM, just one small suggestion

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - looks good to me 👍

File: webapp/channels/src/utils/constants.tsx [hunk 2]
```diff
@@ -1391,6 +1391,18 @@ export const DefaultRolePermissions = {
     ],
 };
 
+// ModeratedPermissions are permissions that can be turned off for members and guests
+// on a per channel basis. These permissions are on by default for team/channel admins.
+export const ModeratedPermissions = [
```
Comment (esarafianou) [inline webapp/channels/src/utils/constants.tsx:1396]: Do we not need to add here the bookmarks permissions that can also be moderated on a per channel basis?
Screenshot 2024-09-09 at 14 59 45

**Review by esarafianou — APPROVED** [review]
Looks good! Just one comment.

File: webapp/channels/src/utils/constants.tsx (hunk 2 above)
Comment (sbishel) [reply webapp/channels/src/utils/constants.tsx:1396]: Yep...missed those.

**Comment by devinbinnie** [comment]
@sbishel Does this need my review? Looks like we've already got 2 dev reviews done.

**Comment by sbishel** [comment]
> @sbishel Does this need my review? Looks like we've already got 2 dev reviews done.
I didn't add you...it happened automatically

**Pull request merged**